### ISSUE TITLE
Bug: OpenAPI - endpoint naming convention has changed significantly (Redoc & Elements).

### ISSUE THREAD

**Issue by chris-telemetry** [starlite-api/starlite#1177]
The endpoint naming convention generated by Starlite in OpenAPI schemas has changed significantly and is much less readable. It appears the `operationId` associated with route handlers is the cause. This is seen in Redoc, and Elements (both of which use operationId to title routes by default) but not Swagger (it only uses the path).

In Redoc/Elements the following handler:
```python
@post("/login")
async def login_handler(request: "Request[Any, Any, Any]", data: User) -> Response[User]:
      pass

@get("/some-path")
def some_route_handler(request: Request[User, Token, Any]) -> Any:
    pass
```

`login_handler` will be named `LoginLoginHandler` whereas it used to simply yield the name `Login Handler`.
`some_router_endpt` will be named `Some-PathSomeRouteHandler`.
If I were to add a router to these it would yield `Some-RouterSome-PathSomePathRouteHandler`.

Setting the `summary='Login Handler` resolves the issue in both Redoc & Elements.

I believe this was done to ensure that the `operationId` is unique to each handler. I support the change as that's an issue I encountered early in development. However we've developed dozens of endpoints with the old convention in mind. 

I'm requesting a change be made in `starlite.openapi.path_item.py` to default the `summary` field to the old convention if no value for `summary` is supplied.

Something to this effect is what I'm thinking:
```python
  operation = Operation(
                operationId=operation_id,
                tags=tags,
                summary=route_handler.summary or unwrap_partial(route_handler.handler_name).title().replace("_", " "),
                description=get_description_for_handler(route_handler, use_handler_docstrings),
                deprecated=route_handler.deprecated,
                responses=create_responses(
                    route_handler=route_handler,
                    raises_validation_error=raises_validation_error,
                    generate_examples=create_examples,
                    plugins=plugins,
                ),
                requestBody=request_body,
                parameters=parameters,  # type: ignore[arg-type]
                security=security,
            )

```

I'm happy to make a PR (cleaned up a bit) if you agree.

**Comment by Goldziher** [comment]
Hi @chris-telemetry. 

Yes this is s known problem. We incorporated an update that was meant to prevent operationId duplications, since these are forbidden in OpenAPI 3.2. the update though required some improvement. Can you test the pending PR that updates this - https://github.com/starlite-api/starlite/pull/1174 and let me know what you think?

**Comment by chris-telemetry** [comment]
Hey @Goldziher,

As always I really appreciate the quick response.

I took a look at the PR and did some testing. I think this PR definitely solves some issues and adds some nice features (love the `operation_id_creator` on OpenAPIConfig).

I think it solves a parallel issue rather than addressing the root issue. I think my problem lies in the fact that both Redoc and Elements use the `operationId` if no value is supplied for `summary`. 

Since the operationId must be unique, it will trend towards being difficult to read.  

Here's what Elements & Redoc look like on your branch:
<img width="1906" alt="Screen Shot 2023-02-08 at 3 48 13 PM" src="https://user-images.githubusercontent.com/78052999/217648129-a79e4282-2a15-4fdf-b812-7dcd3753a710.png">
<img width="1912" alt="Screen Shot 2023-02-08 at 3 52 12 PM" src="https://user-images.githubusercontent.com/78052999/217648392-4594de8a-224c-4bb8-a0d9-8c9357f1752a.png">

I'm suggesting that we default the value of `summary` similar to how we default the value of `operationId`.

So we can get something that looks like this in Elements & Redoc:

<img width="1043" alt="Screen Shot 2023-02-08 at 3 54 59 PM" src="https://user-images.githubusercontent.com/78052999/217648933-763af8d9-d573-46c2-bbd6-2b59548dfd85.png">


If you believe this to be too much of an edge case to fix that's alright. The `operation_id_creator` in your new branch provides me with the opportunity for a serviceable workaround.

```python
def operation_id_creator(handler: HTTPRouteHandler, http_method: Any, path_components: Any) -> str:
    handler.summary = handler.handler_name.title().replace('_', ' ')
    return default_operation_id_creator(handler, http_method, path_components)
```

**Comment by Goldziher** [comment]
@chris-telemetry no problem, we can easily address this. What would be your preferred solution? Easiest for me would be to just default summary to equal the callable name normalized into camelCase, but im open to other ideas of course.

**Comment by chris-telemetry** [comment]
I think the callable name in camelCase or TitleCase would be perfect.

Really appreciate the help.

**Comment by Goldziher** [comment]
@chris-telemetry - PR added. It will be released in v2.0 on the 20th of this month. You're welcome to test it.

**Comment by chris-telemetry** [comment]
@Goldziher Tested it out and it looks great to me! Thank you.

Stoked for 2.0 of starlite!
